til-domain/src/main/java/com/til/application/problem/ProblemService.java

```diff
@@ -0,0 +1,29 @@
+package com.til.application.problem;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.til.domain.problem.dto.ProblemInfoDto;
+import com.til.domain.problem.model.Problem;
+import com.til.domain.problem.repository.ProblemRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class ProblemService {
+
+    private final ProblemRepository problemRepository;
+
+    @Transactional(readOnly = true)
+    public List<ProblemInfoDto> getProblemList() {
```

Comment:
```suggestion
@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class ProblemService {

    private final ProblemRepository problemRepository;
    
    public List<ProblemInfoDto> getProblemList() {
```

P2: @Transactional(readOnly = true)를 서비스 레벨로 빼주고, 트랜잭션을 사용하는 메소드에 @Transactional을 따로 명시하는 방식은 어떠신가요?

Reply:
넵 이 부분 조금 고민했는데, 제안해주신 방법이 더 좋을 것 같습니다! 
수정해놓겠습니다!